### PR TITLE
Fix incorrect API key causes unexpected exception

### PR DESCRIPTION
**Pull request by ezesplit**
When using an invalid API key the SDK raises a `NoMethodError: undefined method `[]' for nil:NilClass` exception that doesn't look too good. This PR addresses that by adding logic to treat this error and log a proper message indicating the possible cause of the issue.

Missing:
- Tests
- Fix Rubocop violations on new code

### CODE REVIEW

File: lib/splitclient-rb/engine/api/segments.rb [hunk 1]
```diff
@@ -10,52 +15,53 @@ def initialize(api_key, config, metrics, segments_repository)
...
           since = @segments_repository.get_change_number(name)
-          while true
-            fetch_segments(name, prefix, since).each { |segment| @segments_repository.add_to_segment(segment) }
-            @config.logger.debug("Segment #{name} fetched before: #{since}, till: #{@segments_repository.get_change_number(name)}") if @config.debug_enabled
+          loop do
+            segment = fetch_segment_changes(name, since)
+            @segments_repository.add_to_segment(segment)
 
-            break if (since.to_i >= @segments_repository.get_change_number(name).to_i)
+            @config.logger.debug("Segment #{name} fetched before: #{since}, \
```
Comment (SenhorCastor) [inline lib/splitclient-rb/engine/api/segments.rb]: no if config.debug.enabled?

Comment (SenhorCastor) [reply lib/splitclient-rb/engine/api/segments.rb]: actually missing log_if_debug?

**Comment by SenhorCastor** [comment]
Also move the version to the pre.rc that we talked

**Comment by ignaciosplit** [comment]
@SenhorCastor feedback applied

**Pull request merged**